core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -929,19 +929,54 @@ def convert_to_refit_full_template(self):
 
         return template
 
-    def hyperparameter_tune(self, hyperparameter_tune_kwargs, hpo_executor=None, time_limit=None, **kwargs):
+    def hyperparameter_tune(self,
+                            hyperparameter_tune_kwargs='auto',
+                            hpo_executor: HpoExecutor = None,
+                            time_limit: float = None,
+                            **kwargs):
+        """
+        Perform hyperparameter tuning of the model, fitting multiple variants of the model based on the search space provided in `hyperparameters` during init.
+
+        Parameters
+        ----------
+        hyperparameter_tune_kwargs : str or dict, default='auto'
+            TODO: Add docstring
+        hpo_executor : HpoExecutor, default None
+            TODO: Add docstring
+        time_limit : float, default None
+            In general, this is the time limit in seconds to run HPO for.
+            In reality, this is the time limit in seconds budget to fully train all trials executed by HPO.
+            For example, BaggedEnsemble will only use a fraction of the time limit during HPO because it needs the remaining time later to fit all of the folds of the trials.
+        **kwargs :
+            Same kwargs you would pass to fit call, such as:
+                X
+                y
+                X_val
+                y_val
+                feature_metadata
+                sample_weight
+                sample_weight_val
+
+        Returns
+        -------
+        Tuple of (hpo_results: Dict[str, dict], hpo_info: Any)
+        hpo_results: Dict[str, dict]
+            A dictionary of trial model names to a dictionary containing:
+                path: str
+                    Absolute path to the trained model artifact. Used to load the model.
+                val_score: float
+                    val_score of the model
+                trial: int
+                    Trial number of the model, starting at 0.
+                hyperparameters: dict
+                    Hyperparameter config of the model trial.
+        hpo_info: Any
+            Advanced output with scheduler specific logic, primarily for debugging.
```

Comment:
In the case of ray tune, this actually is the analysis/result of the experiment: https://docs.ray.io/en/latest/tune/api_docs/analysis.html. Maybe we can mention this in the doc as well and have a pointer to their doc site

Reply:
We can merge this PR and then you could open a PR that refines the docs since you are more familiar, or you can directly edit this PR branch with the doc improvements.